src/fontra/views/editor/scene-controller.js

```diff
@@ -1333,6 +1371,75 @@ function getSelectedContours(path, pointSelection) {
   return [...selectedContours];
 }
 
+function getSelectedOpenContours(
+  path,
+  pointSelection,
+  ignoreSinglePointContour = true
+) {
+  if (!path || !pointSelection) {
+    return [];
+  }
+  const selectedContours = new Set();
+  for (const contourIndex of getSelectedContours(path, pointSelection)) {
+    if (path.contourInfo[contourIndex].isClosed) {
+      // skip if contour is closed already
+      continue;
+    }
+    const contour = path.getContour(contourIndex);
+    const numOnCurvePoints = contour.pointTypes.reduce(
+      (acc, pointType) =>
+        acc +
+        ((pointType & VarPackedPath.POINT_TYPE_MASK) === VarPackedPath.ON_CURVE
+          ? 1
+          : 0),
+      0
+    );
+    if (numOnCurvePoints === 1 && ignoreSinglePointContour) {
+      // skip single point contour
+      continue;
+    }
+    selectedContours.add(contourIndex);
+  }
+
+  return [...selectedContours];
+}
+
+function closeContourEnsureCubicOffCurves(path, contourIndex) {
+  const contourInfo = path.contourInfo[contourIndex];
+  const numContourPoints = path.getContour(contourIndex).pointTypes.length;
+
+  const prevEndPoint = path.getPoint(contourInfo.endPoint - 1);
+  const endPoint = path.getPoint(contourInfo.endPoint);
+  const startPoint = path.getPoint(contourInfo.endPoint - numContourPoints + 1);
+
+  if (startPoint.type && !endPoint.type) {
+    // contour starts with off-curve and ends with on-curve point,
+    // reverse the contour, first.
```

Comment:
You can't just reverse the winding direction behind the user's back... Please find a better solution.

Reply:
OK, I am happy to find a different solution. But I am now wondering if I misunderstood your feedback? Because you wrote this: _Reverse the contour then "close contour" should work just the same._

Reply:
Ah, I can see how that can be misinterpreted. What I meant was: "if you have an open contour with a dangling off curve point at the end, and then you (the user) reverse it, 'close open contour' should also work". That was a roundabout way of saying "if the dangling off-curve point is at the _start_ of the open contour, 'close open contour' should also work."

Reply:
Makes of cause totally sense. Sorry for that misunderstanding. 
I pushed a solution. Here a short demo of different open paths:

https://github.com/googlefonts/fontra/assets/21055547/5de96654-8b8c-4cb4-9eca-c05e9b280270

I won't close this review/feedback, because I am not sure if you will like the solution.